packages/@react-stately/table/src/useColumnResizeWidthState.ts

```diff
@@ -0,0 +1,227 @@
+
```

Comment:
Some of this is functionality pulled out from TableLayout and some is functionality that we added. 

If you would like to see a simplified version of the algorithm we wanted to use here then you can view this:
https://codesandbox.io/s/column-width-resizer-thingy-danny-fork-jo4rr?file=/src/column-utils.js

You can note the algorithm with comments and the spec file